### PR TITLE
Rewrite toolStrip and richTextBox MAUI tests

### LINKED ISSUE #3617: MAUI tests fail frequently and aren't easily diagnosable
These are an attempt to shim in the old internal tests (note that MAUI here is an old internal test framework). The old tests worked quite differently and don't fit in well with the current test infrastructure. We're getting [frequent intermittent failures](https://dev.azure.com/dnceng/public/_build/results?buildId=737706&view=ms.vss-test-web.build-test-results-tab&runId=22844842&resultId=100000&paneView=history) like so:

```
System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(scenarioName: \"Hotkey_Fires_OnClick\")

Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Pass\r\nActual:   Fail\r\n          ↑ (pos 0)

Stack trace
   at System.Windows.Forms.Maui.IntegrationTests.MauiTestHelper.ValidateScenarioPassed(String projectName, String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/MauiTestHelper.cs:line 90
   at System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiButtonTests.cs:line 35
``` 

This doesn't really help you much and it isn't obvious as to what you can do to move forward. In addition, the MAUI infrastructure sources aren't available so it is difficult to understand how the whole system works.

My recommendation (and current POR):

1. Disable the MAUI tests for now
2. Make them more xUnit-like and have any shim code in the repo
3. If we can make the tests plug in as a "normal" xUnit test then re-enable them for the CI, otherwise leave them for manual runs

cc: @AdamYoblick, @RussKie 

Failed with #3600

### PR DESCRIPTION
**Pull request by Danil-Andrianov**
Fixes #3617

## Proposed changes

- Replaced toolStripMAUI tests into unit tests project
- Replaced richTextBoxMAUI tests into UI IntegrationTests project

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6396)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs [hunk 1]
```diff
@@ -7176,6 +7176,61 @@ public void ToolStrip_WndProc_InvokeMouseHoverWithHandle_Success()
             Assert.Equal(0, createdCallCount);
         }
 
+        [WinFormsFact]
+        public void ToolStrip_KeyboardAccelerators_ReturnsExpected()
+        {
+            SubToolStripDropDown toolStrip = new ();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs:7182]:
```suggestion
            using SubToolStripDropDown toolStrip = new ();
```

**Comment by dreddy-work** [comment]
LGTM.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+                    // Let the event queue drain after clicking, before moving the cursor back to the old position.
+                    Application.DoEvents();
+                }
+                finally
+                {
+                    richTextBox.LinkClicked -= handler;
+
+                    if (setOldCursorPos.IsTrue())
+                    {
+                        // Move cursor to old position
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs:145]:
```suggestion
                        // Move cursor to the old position.
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs:221]:
```suggestion
                        // Move cursor to the old position.
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs:69]:
```suggestion
                        // Move cursor to the old position.
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+{\*\generator Riched20 10.0.17134}\viewkind4\uc1 
+{\field{\*\fldinst { HYPERLINK ""http://www.google.com"" }}{\fldrslt {Click link #1}}}
+\pard\sa200\sl276\slmult1\f0\fs22\lang9  for more information.\par
+{\field{\*\fldinst { HYPERLINK ""http://www.google.com"" }}{\fldrslt {Click link #2}}}
+\pard\sa200\sl276\slmult1\f0\fs22\lang9  for more information.\par
+{\field{\*\fldinst { HYPERLINK ""http://www.google.com"" }}{\fldrslt {Click link #3}}}
+\pard\sa200\sl276\slmult1\f0\fs22\lang9  for more information.\par
+}";
+
+                Point previousPosition = new Point();
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs:38]:
```suggestion
                Point previousPosition = new();
```
or:
```suggestion
                Point previousPosition = default;
```
Ditto below.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs [hunk 6]
```diff
@@ -7176,6 +7176,65 @@ public void ToolStrip_WndProc_InvokeMouseHoverWithHandle_Success()
             Assert.Equal(0, createdCallCount);
         }
 
+        [WinFormsFact]
+        public void ToolStrip_KeyboardAccelerators_ReturnsExpected()
+        {
+            using SubToolStripDropDown toolStrip = new ();
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs:7182]:
```suggestion
            using SubToolStripDropDown toolStrip = new();
```

**Review by WPMGPRoSToTeMa — APPROVED** [review]
👍

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs (hunk 5 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/RichTextBoxTests.cs:38]: I think that first suggestion would be more suitable, thank you.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**